### PR TITLE
feat: Added keyword 'subclassof'

### PR DESCRIPTION
**Pull request by philippefutureboy**
Added keyword subclassof, a specific case of instanceof used to validate whether the passed argument extends the constructor used in the schema.

I made this keyword a year ago, but never contributed it back to the main repository. About time :)

Cheers 🚀

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -95,13 +96,13 @@ The value of the keyword should be a string (`"Object"`, `"Array"`, `"Function"`
...
 
-```
+```javascript
 ajv.validate({ instanceof: 'Array' }, []); // true
 ajv.validate({ instanceof: 'Array' }, {}); // false
 ajv.validate({ instanceof: ['Array', 'Function'] }, function(){}); // true
 ```
 
-You can add your own constructor function to be recognised by this keyword:
+You can add your own constructor function to be recognized by this keyword:
```
Comment (epoberezkin) [inline README.md]: "recognised" is the British spelling :)

File: README.md [hunk 2]
```diff
@@ -112,6 +113,38 @@ instanceofDefinition.CONSTRUCTORS.MyClass = MyClass;
 ajv.validate({ instanceof: 'MyClass' }, new MyClass); // true
 ```
 
+### `instanceof`
```
Comment (epoberezkin) [inline README.md]: Should it be `subclassof`?

File: README.md [hunk 3]
```diff
@@ -112,6 +113,38 @@ instanceofDefinition.CONSTRUCTORS.MyClass = MyClass;
...
+```
+
+You can add your own constructor function to be recognized by this keyword:
+
+```javascript
+class MyClass {}
+class MyChildClass extends MyClass {}
+var subclassofDefiniton = require('ajv-keywords').get('subclassof');
+// or require('ajv-keywords/keywords/subclassof');
+var ajv = defFunc(new Ajv(), { MyClass: MyClass });
```
Comment (epoberezkin) [inline README.md]: It seems incorrect

File: keywords/subclassof.js [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+    Object: Object,
+    Array: Array,
+    Function: Function,
+    Number: Number,
+    String: String,
+    Date: Date,
+    RegExp: RegExp,
+    Error: Error
+  };
+
```
Comment (epoberezkin) [inline keywords/subclassof.js]: instanceof also adds Buffer is it's available - probably should be here as well

File: keywords/subclassof.js [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+    });
+  }
+
+  var definition = {
+    compile: function (schema) {
+      if (typeof schema == 'string') {
+        var Constructor = getConstructor(schema);
+        return function (data) {
+          return data.prototype instanceof Constructor ||
+            (data.constructor.prototype instanceof Constructor && data !== Object);
```
Comment (epoberezkin) [inline keywords/subclassof.js]: `data.constructor.prototype instanceof Constructor` is probably the same as `data instanceof Constructor`, I don't think that instances should pass this keyword, only subclasses should.

File: README.md (hunk 1 above)
Comment (philippefutureboy) [reply README.md]: Ah! Will revert that change :P

File: README.md (hunk 3 above)
Comment (philippefutureboy) [reply README.md]: Woops. Changing to `subclassofDefiniton`

File: keywords/subclassof.js (hunk 4 above)
Comment (philippefutureboy) [reply keywords/subclassof.js]: Yup, it was related to the fact that subclassof was validating against instances too. Buffer is a final class, and cannot be extended. But now that I'll modify this to be orthogonal to instanceof, it won't be a problem anymore.

**Comment by epoberezkin** [comment]
Closing as old.

**Pull request closed without merging**